src/policy/space.rs

```diff
@@ -229,7 +219,16 @@ pub trait Space<VM: VMBinding>: 'static + SFT + Sync + Downcast {
 
     fn address_in_space(&self, start: Address) -> bool {
         if !self.common().descriptor.is_contiguous() {
-            self.common().vm_map().get_descriptor_for_address(start) == self.common().descriptor
+            // If the space is not contiguous, we use `SFT_MAP` to check.
+            //
+            // Note: When comparing references (`&T`) using `==`, it compares by values.
+            // To compare by identities, we need to compare pointers.
+            //
+            // TODO: For `DenseChunkMap`, it is possible to reduce one level of dereferencing by
+            // letting each space remember its index in `DenseChunkMap::index_map`.
+            let self_ptr = self as *const _ as *const ();
```

Comment:
We could add a method in `SFT` to return the space descriptor, and compare on that.

Reply:
That would be less efficient. The current code is good.